pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -98,6 +98,11 @@ func getPodName(set *apps.StatefulSet, ordinal int) string {
 // must be a PersistentVolumeClaim from set's VolumeClaims template.
 func getPersistentVolumeClaimName(set *apps.StatefulSet, claim *v1.PersistentVolumeClaim, ordinal int) string {
 	// NOTE: This name format is used by the heuristics for zone spreading in ChooseZoneForVolume
+	var volumeClaimBaseName string
+	volumeClaimBaseName = claim.ObjectMeta.Annotations["volumeClaimBaseName"]
```

Comment:
What happens when this annotation is changed after the StatefulSet and its pods are created? Volume claim names should be immutable.